libnetwork/drivers/bridge/internal/nftabler/network.go

```diff
@@ -4,29 +4,261 @@ package nftabler
 
 import (
 	"context"
+	"fmt"
 
 	"github.com/containerd/log"
+	"github.com/docker/docker/internal/cleanups"
 	"github.com/docker/docker/libnetwork/drivers/bridge/internal/firewaller"
+	"github.com/docker/docker/libnetwork/internal/nftables"
+	"go.opentelemetry.io/otel"
 )
 
 type network struct {
-	config firewaller.NetworkConfig
-	fw     *nftabler
+	config  firewaller.NetworkConfig
+	cleaner func(ctx context.Context) error
+	fw      *nftabler
 }
 
 func (nft *nftabler) NewNetwork(ctx context.Context, nc firewaller.NetworkConfig) (_ firewaller.Network, retErr error) {
 	n := &network{
 		fw:     nft,
 		config: nc,
 	}
+	ctx = log.WithLogger(ctx, log.G(ctx).WithFields(log.Fields{"bridge": n.config.IfName}))
+
+	var cleaner cleanups.Composite
+	defer func() {
+		if err := cleaner.Call(ctx); err != nil {
+			log.G(ctx).WithError(err).Warn("Failed to clean up nftables rules for network")
+		}
+	}()
+
+	if n.fw.config.IPv4 {
+		clean, err := n.configure(ctx, nft.table4, n.config.Config4)
+		if err != nil {
+			return nil, err
+		}
+		cleaner.Add(clean)
```

Comment:
You need to add a nil check here and below, otherwise a nil func will be called by `cleanups.Composite` and that will cause a panic.